src/NServiceBus.RabbitMQ.Tests/ConnectionString/ConnectionStringParserTests.cs

```diff
@@ -44,7 +44,7 @@ public void Should_fail_if_host_is_not_present()
         {
             var parser = new ConnectionStringParser(settings);
 
-            Assert.Throws<Exception>(() => parser.Parse("virtualHost=Copa;username=Copa;password=abc_xyz;port=12345;requestedHeartbeat=3"));
+            Assert.Throws<NotSupportedException>(() => parser.Parse("virtualHost=Copa;username=Copa;password=abc_xyz;port=12345;requestedHeartbeat=3"));
```

Comment:
I'm not sure [`NotSupportedException`](https://msdn.microsoft.com/en-us/library/system.notsupportedexception(v=vs.110).aspx) is appropriate. On face value it seems to be, and I often reach for it myself before the help suggests otherwise:

> The exception that is thrown when an invoked method is not supported, or when there is an attempt to read, seek, or write to a stream that does not support the invoked functionality.

That said, I can't think of a better one right now. I usually default to `InvalidOperationException` when nothing else fits. 😄